src/sentry/static/sentry/app/views/eventsV2/table/columnEditModal.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+
+import Button from 'app/components/button';
+import ButtonBar from 'app/components/buttonBar';
+import {DISCOVER2_DOCS_URL} from 'app/constants';
+import {ModalRenderProps} from 'app/actionCreators/modal';
+import {t} from 'app/locale';
+import {OrganizationSummary} from 'app/types';
+import {Column} from '../eventView';
+import ColumnEditCollection from './columnEditCollection';
+
+type Props = {
+  columns: Column[];
+  organization: OrganizationSummary;
+  tagKeys: string[];
+  // Fired when column selections have been applied.
+  onApply: (columns: Column[]) => void;
+} & ModalRenderProps;
+
+type State = {
+  columns: Column[];
+};
+
+class ColumnEditModal extends React.Component<Props, State> {
+  state = {
+    columns: this.props.columns,
+  };
+
+  handleChange = (columns: Column[]) => {
+    this.setState({columns});
```

Comment:
@markstory Later on when this component is integrated into Discover2; if there is only one column, we'll want to ensure the user is unable to delete the remaining column.

Reply:
I'll add a subtask to asana for this.